### ISSUE TITLE
Potential bug in arc.env()$workspace READ-ONLY on server deployments

### ISSUE THREAD

**Issue by f-tonini** [R-ArcGIS/r-bridge#17]
In a Desktop architecture, it seems like arc.env()$workspace can be correctly used as a scratch workspace to write output files. However, when the same R script is published as a GP service and deployed on a Server architecture, arc.env()$workspace becomes READ-ONLY and throws errors (see below) when trying to write to that location:

“Error executing tool. Factor Analysis for Mixed Data Job ID: jac57dc60e745433bb91359a45e0b2bcc : Error in sink : invalid 'description' argument? Failed to execute (FAMD).? Failed to execute (Factor Analysis for Mixed Data)."

message:"Unable to complete operation."

This only happens on a server deployment, as the same script works correctly on Desktop. The function sink() is an R command to write to an output text file.

**Comment by ChristopherTracey** [comment]
Has anyone come up with a solution to this issue? We're hitting it as well with outputting a pdf file generated by a r script.

**Comment by scw** [comment]
@ChristopherTracey we are aware of the issue, but haven't created a fix yet, I will keep this issue up to date once we do.

**Comment by ChristopherTracey** [comment]
Thanks. Do you know of any potential short-term workarounds?-We're trying to get our project up and running in the the coming weeks.

On Dec 19, 2017, at 5:41 PM, Shaun Walbridge <notifications@github.com<mailto:notifications@github.com>> wrote:


@ChristopherTracey<https://github.com/christophertracey> we are aware of the issue, but haven't created a fix yet, I will keep this issue up to date once we do.

-

**Comment by scw** [comment]
Can you write the PDF out to another directory with permissive permissions set on it?

**Comment by ChristopherTracey** [comment]
Yes, but we've been unable to move it to the WebApp's scratchFolder to get it to display in the tool output.

On Dec 19, 2017, at 5:49 PM, Shaun Walbridge <notifications@github.com<mailto:notifications@github.com>> wrote:


Can you write the PDF out to another directory with permissive permissions set on it?

-

**Comment by f-tonini** [comment]
Everyone,

This is the same issue I had. You can write to a different folder, yes, but not the scratch folder which is much needed for the tasks I am running.

**Comment by ChristopherTracey** [comment]
We developed a workaround for this (emailing the results to the user instead of opening up in the browser), but we look forward to the fix for this for other uses of the bridge.
